fog/overseer/server/tests/inactive_outstanding_key_idle_has_key.rs

```diff
@@ -29,153 +19,72 @@ const PORT_NUMBER: u16 = 8080;
-        peer_indices.iter().cloned(),
-        recovery_db.clone(),
-        &watcher_path,
-        &ledger_db_path,
-        logger.clone(),
-    );
-    let (node2, _state_file_node2, client_listen_uri2) = utils::make_node(
-        2,
-        peer_indices.iter().cloned(),
-        recovery_db.clone(),
-        &watcher_path,
-        &ledger_db_path,
-        logger.clone(),
-    );
+    let mut ctx = utils::TestContext::new(BASE_PORT, true, logger.clone());
+    let mut nodes = VecDeque::from(ctx.make_nodes(3));
 
-    assert!(!node0.is_active());
-    assert!(!node1.is_active());
-    assert!(!node2.is_active());
+    assert!(!nodes[0].is_active());
+    assert!(!nodes[1].is_active());
+    assert!(!nodes[2].is_active());
 
     // Give RPC etc. time to start
-    std::thread::sleep(Duration::from_secs(1));
+    sleep(Duration::from_secs(1));
 
-    node0.activate().expect("node0 failed to activate");
+    nodes[0].activate().expect("nodes[0] failed to activate");
 
-    assert!(node0.is_active());
-    assert!(!node1.is_active());
-    assert!(!node2.is_active());
+    assert!(nodes[0].is_active());
+    assert!(!nodes[1].is_active());
+    assert!(!nodes[2].is_active());
 
-    let first_node0_ingress_key = node0.get_ingest_summary().get_ingress_pubkey().clone();
-    let first_node1_ingress_key = node1.get_ingest_summary().get_ingress_pubkey().clone();
-    let first_node2_ingress_key = node2.get_ingest_summary().get_ingress_pubkey().clone();
+    let first_node0_ingress_key = nodes[0].get_ingest_summary().get_ingress_pubkey().clone();
+    let first_node1_ingress_key = nodes[1].get_ingest_summary().get_ingress_pubkey().clone();
+    let first_node2_ingress_key = nodes[2].get_ingest_summary().get_ingress_pubkey().clone();
 
     // All the keys should be equal.
     assert_eq!(first_node0_ingress_key, first_node1_ingress_key);
     assert_eq!(first_node0_ingress_key, first_node2_ingress_key);
     assert_eq!(first_node1_ingress_key, first_node2_ingress_key);
 
-    // Initialize an OverSeerService object
-    let mut overseer_service = OverseerService::new(
-        vec![client_listen_uri0, client_listen_uri1, client_listen_uri2],
-        recovery_db.clone(),
-        logger.clone(),
-    );
-    overseer_service.start().unwrap();
-
-    // Set up the Rocket instance
-    let overseer_state = OverseerState { overseer_service };
-    // TODO: Consider testng the CLI here instead
-    let rocket_config = rocket::Config::figment()
-        .merge(("port", PORT_NUMBER))
-        .merge(("address", "127.0.0.1"));
-    let rocket = server::initialize_rocket_server(rocket_config, overseer_state);
-    let client = Client::tracked(rocket).expect("valid rocket instance");
+    // Initialize an OverseerService with an associated server.
+    let client = ctx.make_overseer_client_for_nodes(nodes.make_contiguous());
     client.post("/enable").dispatch();
+    // Give overseer time to perform its logic.
+    sleep(Duration::from_secs(10));
```

Comment:
This sleep duration is new while I understand there may have been a time lapse miss, 10 seconds seems like a lot.

THe old code used to call `overseer_service.start().unwrap()`, but I'm not seeing that in the new implementation.  
Could that be the cause for the delay?  Previously it would block on `start()`, but now the rocket server has to start the overseer in its thread?

Reply:
No, that call now happens from `make_node()`

Reply:
I misspoke, it now happens in `enable_overseer()`